File: solutions/python/series/1/series.py
# Solely in honor of Shree DR.MDD

def slices(series, length):
  if length == 0:
    raise ValueError('slice length cannot be zero')
  if length < 0:
    raise ValueError('slice length cannot be negative')
  if not series:
    raise ValueError('series cannot be empty')
  if length > len(series):
    raise ValueError('slice length cannot be greater than series length')

  bundle = []
  for idx in range(len(series) - length + 1):
    bundle.append(series[idx:idx + length])
  return bundle

# vim:ts=2:sw=2:expandtab

File: solutions/javascript/series/1/series.js
// For my Shree DR.MDD

export class Series {
  constructor(numberSeries) {
    if(numberSeries === ''){
      throw new Error('series cannot be empty');
    }
    this._series = numberSeries;
  }

  slices(lengthOfSlice) {
    if(lengthOfSlice === 0){
      throw new Error('slice length cannot be zero');
    } else if(lengthOfSlice < 0){
      throw new Error('slice length cannot be negative');
    } else if(this._series.length < lengthOfSlice){
      throw new Error('slice length cannot be greater than series length');
    }

    const resultSlices = [];
    for(let start = 0; start <= (this._series.length - lengthOfSlice); start++){
      const sliceArray = [];
      for(let offset = 0; offset < lengthOfSlice; offset++){
        sliceArray.push(Number(this._series[start + offset]));
      }
      resultSlices.push(sliceArray);
    }

    return resultSlices;
  }
}
